malcolm/modules/ADCore/parts/detectordriverpart.py

```diff
@@ -49,16 +100,27 @@ def reset(self, context):
     @add_call_types
     def report_status(self):
         # type: () -> UInfos
+        ret = []
         if self.main_dataset_useful:
-            return NDArrayDatasetInfo(rank=2)
+            ret.append(NDArrayDatasetInfo(rank=2))
+        for _, attrInd in enumerate(self.extra_attributes.value.sourceId):
```

Comment:
I'd probably do:
`for name, dataset_type in zip(self.extra_attributes.value.name, self.extra_attributes.value.datasetType):`